core/src/program/blocks/mod.rs

```diff
@@ -35,9 +36,9 @@ impl CodeBlock {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a new Span block instantiated with the provided operations.
-    pub fn new_span(operations: Vec<Operation>) -> Self {
-        Self::Span(Span::new(operations))
+    /// Returns a new Span block instantiated with the provided operations and decorator list.
+    pub fn new_span(operations: Vec<Operation>, decorators: DecoratorList) -> Self {
+        Self::Span(Span::new(operations, decorators))
```

Comment:
Maybe rather than changing this method signature we should add another method - e.g., `new_span_with_decorators()`? I think this may reduce the number of places where we need to make updates.

Reply:
Done. Initially I implemented a similar method to keep the changes minimal but then it seemed a little redundant so removed it.